### ISSUE TITLE
“Download” link of concepts list does not give glosses/concepticon, only names.

### ISSUE THREAD

**Issue by Anaphory** [evolaemp/northeuralex_website#1]
The XLS download adapter for the concepts list (http://northeuralex.org/parameters.xls) gives only the ID and Name column of the parameters. Glosses and concepticon IDs are not included, but would be nice to get.

**Comment by Anaphory** [comment]
The CSV download is very complete, but the sorting of columns is not obvious.

**Comment by pavelsof** [comment]
Fixed! Now the xls and csv downloads share the same columns for each of these data tables: languages, concepts, and words. Also, the json downloads (which seemed to be useless) are fixed to be arrays of objects where the keys are the same as the csv/xls columns.

**Comment by xrotwang** [comment]
Yeah, the default adapters for DataTables don't do much, except listing object IDs and URLs. @pavelsof glad to see you figured out the recommended way to fix this!

**Comment by pavelsof** [comment]
@xrotwang, thank you! I just copied what I saw in `clld/web/adapters/excel.py` (and I only spared the other formats apart from csv and json because I am not sure about their use cases).

Perhaps for the clld project it might be a good idea to refactor the `clld/web/adapters/*` modules in a way similar to the `excel` module, so that clld apps do not have to implement their own `CsvAdapter`, `JsonAdapter`, etc.

**Comment by xrotwang** [comment]
At least regarding the CSV adapters, I hope these will be superseded by a complete download of a dataset in [CLDF](https://github.com/glottobank/cldf). In general, I think the download menu on each DataTable is more misleading than useful, because what can be downloaded typically does not mirror what is displayed in the HTML table (mostly for performance reasons). The main reason I implemented this functionality is that WALS Online had multiple download formats per feature, and I thought I could make these available in that way.

But these days, I'd say anyone trying to work quantitatively with a clld dataset should get the whole database (and be able to slice and dice it as necessary themselves, in R, python, or Excel for that matter).
